onnx/helper.py

```diff
@@ -311,6 +311,23 @@ def make_map(
     return map
 
 
+def make_optional(
+        name,   # type: Text
+        elem_type,   # type: int
```

Comment:
The annotation ```type: int``` doesn't sound correct.

Reply:
Could be a pre-existing problem with the Sequence also

Reply:
A couple of related questions: (a) Does the caller have to specify elem_type, or should we infer it from the actual value? (b) What is the purpose of elem_type in the proto, is it to just indicate which variant is being used or to indicate the full-type of the contained value? I think we can only do the former, not the latter. Once the "proto" spec is clarified, I think the helper methods can be adjusted accordingly.

Reply:
Stated another way: how is a None value represented in the Proto? Is elem_type supposed to be UNDEFINED?

Reply:
@gramalingam 
I think from what we discussed, we decided to mandate setting the elem_type even for None.
Use of UNDEFINED I think is mainly for data serialization purposes.

Reply:
another question regarding the elem_type - similar to Rama's question what is the expected value for elem_type? Is it the type held by Optional say Optional <Tensor, Sequence etc> or the complete type meaning optional(Tensor(float32))